server/src/main/java/io/druid/server/router/TieredBrokerHostSelector.java

```diff
@@ -308,15 +309,15 @@ Server pick()
     {
       ImmutableList<Server> currNodes = nodes;
 
+      int index = roundRobinIndex.getAndIncrement();
```

Comment:
It will produce negative results when index overflows, suggested to replace with a method which wraps around `currNodes.size()` explicitly, with use of compareAndSet(), for sanity

Reply:
+1

Reply:
Or allow the overflow, but make sure the chosen index is positive (use absolute value).